### PR TITLE
[TM] Adds new ruin on lavaland with ghost-role spawners.

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

Adds tech cult base on the lavaland, alongside with Energy Sniper rifle, unique to it.

## Why It's Good For The Game

Ruins are always interesting for miners/explorers.
You will be able to research more weapons, more powerful cybernetic organs and implants if you unlock the Marsian Technology by using one of their items with destructive analyzer(similar to unlocking alien/illegal tech).
Leader of this cult also counts as "holy-role". That means he might be useful as the chaplain on the station if you convince them to help you.

lavaland_surface_techcult-1

## Changelog
:cl:
add: Added Tech Cult base on the lavaland. It contains 5 ghost-roles, new tier4 cybernetic organs, weapons and implants. The base itself is a combination of R&D, Surgery and Chapel.
add: Added hidden tech-node for all of the tech cult technology. To unlock it, you simply put one of their items into the destructive analyzer and use it. Similar to unlocking alien/illegal tech.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
I'm glad you found a place that likes your stuff egor, the drama was painful. 

Update your changelog too

**Comment by EgorDinamit** [comment]
I will update it a bit later, I just placed it here so people will see the main code and maybe idea of this ruin.

**Comment by ATH1909** [comment]
oh boy another lavaland loot pinata

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+	name = "blessed tech robes"
+	desc = "From the rage of the beast, Machine God protect us."
+	armor = list("melee" = 65, "bullet" = 40, "laser" = 40, "energy" = 40, "bomb" = 40, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 90)
+	hoodtype = /obj/item/clothing/head/hooded/techpriest/armor/lead
+
+/obj/item/clothing/suit/hooded/techpriest/armor/lead/Initialize()
+	. = ..()
+	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, ITEM_SLOT_OCLOTHING)
+
+/obj/item/clothing/head/hooded/techpriest/armor/lead
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm:104]: er, this has the exact same typepath as the robes

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+
+//Basic
+/obj/item/clothing/suit/hooded/techpriest/armor
+	name = "armored techpriest robes"
+	desc = "An armored version of robes worn by followers of the machine god."
+	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/pickaxe, /obj/item/organ/regenerative_core/legion, /obj/item/kitchen/knife/combat/bone, /obj/item/kitchen/knife/combat/survival)
+	armor = list("melee" = 45, "bullet" = 10, "laser" = 25, "energy" = 25, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60)
+	hoodtype = /obj/item/clothing/head/hooded/techpriest/armor
+
+/obj/item/clothing/head/hooded/techpriest/armor
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm]: same problem here as below

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+	name = "Adept of the Machine Cult"
+	roundstart = FALSE
+	random = TRUE
+	death = FALSE
+	show_flavour = TRUE
+	mob_name = "tech priest"
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper"
+	short_desc = "You are a member of the machine cult on Lavaland."
+	flavour_text = "The flesh is weak and humans are fragile. You exist only to research universe and enchance abilities with power of science."
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm]:
```suggestion
	flavour_text = "The flesh is weak and humans are fragile. You exist only to research the universe and enchance your abilities with the power of science."
```

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+	important_info = "Listen to your leader, help those in need and protect your religion."
+	outfit = /datum/outfit/techcult
+	assignedrole = "Tech Priest"
+
+/obj/effect/mob_spawn/human/techcult/special(mob/living/new_spawn)
+	new_spawn.grant_language(/datum/language/machine, TRUE, TRUE, LANGUAGE_MIND) //They have to implant robotic box instead of a tongue to speak this language though.
+
+/datum/outfit/techcult/post_equip(mob/living/carbon/human/H)
+	H.faction |= "Mechanicus"
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm]: make sure that their armor won't reject them before they receive the mechanicus faction

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+
+/***************** Spell *****************/
+
+/obj/item/book/granter/spell/omnissiah
+	name = "Misterious book"
+	desc = "The sacred texts, allowing a higher tech priests to directly communicate with what is believed to be the Machine God itself."
+	spell = /obj/effect/proc_holder/spell/aoe_turf/conjure/tech
+	spellname = "holy gift"
+	icon_state ="bookcharge"
+	remarks = list("But will it make me a mech?", "I could've just use the exosuit fabricator, huh?", "All powerful god, send me a honkbot!", "So it is a pocket robotic factory.", "This page is just full of binary code...", "Can it give me a laser arm?", "But can I get combat implants from here?")
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm]:
```suggestion
	remarks = list("But will it make me a mech?", "I could've just used the exosuit fabricator, huh?", "All-powerful god, send me a honkbot!", "So it is a pocket robotics factory.", "This page is just full of binary code...", "Can it give me a laser arm?", "But can I get combat implants from here?")
```

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 6]
```diff
@@ -0,0 +1,592 @@
...
+/obj/item/organ/heart/cybernetic/tier4
+	name = "quadro-cybernetic heart"
+	desc = "An electronic device designed to mimic the functions of an organic human heart. Can inject with set of chemicals when user enter critical condition and regenerate the dose later."
+	icon = 'icons/Fulpicons/tier4_organs.dmi'
+	icon_state = "heart-c-u3"
+	maxHealth = 3 * STANDARD_ORGAN_THRESHOLD
+	dose_available = TRUE
+	emp_vulnerability = 5
+
+/obj/item/organ/heart/cybernetic/tier4/on_life()
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm:433]: shouldn't this already ben inherited from normal cybernetic hearts?

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 7]
```diff
@@ -0,0 +1,592 @@
...
+	desc = "A very dangerous tool that is able to synthesize even more dangerous chemicals."
+	recharge_time = 30 //10u every minute
+
+/obj/item/reagent_containers/borghypo/hacked/augment/regenerate_reagents()
+	for(var/i in 1 to reagent_ids.len)
+		var/datum/reagents/RG = reagent_list[i]
+		if(RG.total_volume < RG.maximum_volume)
+			RG.add_reagent(reagent_ids[i], 5)
+
+/obj/item/reagent_containers/borghypo/hacked/augment/add_reagent(datum/reagent/reagent)
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm:555]: check to see if the parent item already has this stuff

File: code/modules/ruins/lavalandruin_code/techcult.dm [hunk 8]
```diff
@@ -0,0 +1,592 @@
...
+//Advanced Surgical Toolset Implant
+/obj/item/organ/cyberimp/arm/surgery/plus
+	name = "advanced surgical toolset implant"
+	desc = "A set of advanced surgical tools hidden behind a concealed panel on the user's arm."
+	contents = newlist(/obj/item/retractor/advanced/augment, /obj/item/surgicaldrill/advanced/augment, /obj/item/scalpel/advanced/augment, /obj/item/surgical_drapes)
+
+/obj/item/organ/cyberimp/arm/surgery/plus/emag_act(mob/user)
+	if(!(locate(/obj/item/reagent_containers/borghypo/hacked/augment) in items_list))
+		to_chat(user, "<span class='notice'>You unlock and hack integrated hypospray located in [src]!</span>") //Oh god oh fuck
+		items_list += new /obj/item/reagent_containers/borghypo/hacked/augment(src)
```
Comment (ATH1909) [inline code/modules/ruins/lavalandruin_code/techcult.dm:531]: jesus christ what the fuck why

I mean, I guess fulp usually doesn't allow murderboning, and getting this requires raiding/visiting a specific lavaland ruin as a traitor (or someone who has access to an emag, at any rate), so it'll be hard to abuse this TOO badly, but still...

Eh, on second thought, you could just use a sleepypen (or an emagged mediborg) anyway, I guess it's okay.

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm:104]: Nope, it's perfectly fine.
"clothing/head", not "clothing/suit"

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm]: Perfectly fine as above.

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 4 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm]: How dare you assume I didn't test it?

Yes, it works.

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 6 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm:433]: This proc is on tier2 and tier 3 only, **as far as I know.**
Additionally, this one is a bit different.

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 7 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm:555]: Parent has this stuff, but it's a shit-code.
Volume variable is never used so I'm calling this to change it from 30 to 10.

File: code/modules/ruins/lavalandruin_code/techcult.dm (hunk 8 above)
Comment (EgorDinamit) [reply code/modules/ruins/lavalandruin_code/techcult.dm:531]: It's quite balanced.
Keep in mind the SLOW regeneration rate and total volume, which is 10 for each toxin.

**Comment by BarrelOfNapkins** [comment]
Looks cool!

**Comment by BarrelOfNapkins** [comment]
NO!!! IT MUST STAY OPEN!!

**Pull request merged**